azurerm/internal/services/sentinel/resource_arm_sentinel_alert_rule_action.go

```diff
@@ -0,0 +1,192 @@
+package sentinel
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/securityinsight/mgmt/2019-01-01-preview/securityinsight"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	logicParse "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/logic/parse"
+	logicValidate "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/logic/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/sentinel/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/sentinel/validate"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+const OperationInsightsRPName = "Microsoft.OperationalInsights"
+
+func resourceArmSentinelAlertRuleAction() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmSentinelAlertRuleActionCreate,
+		Read:   resourceArmSentinelAlertRuleActionRead,
+		Delete: resourceArmSentinelAlertRuleActionDelete,
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.SentinelAlertRuleActionID(id)
+			return err
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"sentinel_alert_rule_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.SentinelAlertRuleID,
+			},
+
+			"logic_app_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: logicValidate.LogicAppWorkflowID,
+			},
+
+			"logic_app_trigger_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
```

Comment:
could we validate this?

Reply:
There is no source telling what pattern should a trigger name be (the `azurerm_logic_app_trigger_custom` resource hasn't specify a validation for its name either). So I'm going to simply use the `validation.StringIsNotEmpty` here.